### PR TITLE
Set client side validation

### PR DESCRIPTION
**Pull request by chiztechnology**
Hello! We just set up a new feature to validate the form using javascript patterns.
Kindly check and tell us if there's something to improve.
Thanks a lot 🙇‍♂️

### CODE REVIEW

File: js/form_validator.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+const EMAIL_REQUIRED = 'Please enter your email';
+const EMAIL_INVALID = 'Please enter a correct email address format !';
+
+form.addEventListener('submit', (event) => {
+  // validate the form
+  const emailValid = validateEmail(form.elements['.email'], EMAIL_REQUIRED, EMAIL_INVALID);
+  if (!emailValid) {
+    event.preventDefault();
+  }
+});
```
Comment (Gopxfs) [inline js/form_validator.js]: As requested by the project requirements:
image

Your form shouldn't allow upper-case letters 😉.
Currently, I was able to send the following message:
image

Without receiving any custom error message. Please make sure to validate the user email.

Tip: This error is being displayed on the console:
image

Make sure to check all the values being passed to your functions, looks like an `undefined` value is going through the `hasValue()` 😄.

**Review by Gopxfs — CHANGES_REQUESTED** [review]
# ♻️ STATUS: REQUIRED CHANGES ♻️

## Hi team👋,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights :rocket:
✔️ No linter errors.
✔️ Great code structure. Well done!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions 🗒️  
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Gopxfs in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by chiztechnology** [comment]
The issue is solved.
Thank you dear

**Review by Gopxfs — APPROVED** [review]
# 🟢 STATUS: APPROVED 🟢

## Hello team👋,

Your project is complete! You followed all the requirements, feel free to merge it! 😄
Congratulations! 🎉 



## Highlights
✔️ Linters are passing successfully.
✔️ Implemented all the changes requested by the previous reviewer. Well done!

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ha-manel** [comment]
# STATUS: INVALID 🔴 
Hi @maherramadan78 
This PR is invalid because it's your partner's PR😓kindly work on your repository and submit YOUR pull request 👍  

Best of luck!

------
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by maherramadan78** [comment]
The exercice is pair Programming, why its invalid?
We work both on the same one

Enviado desde Correo<https://go.microsoft.com/fwlink/?LinkId=550986> para Windows

De: Manel ***@***.***>
Enviado: viernes, 18 de noviembre de 2022 11:52 a. m.
Para: ***@***.***>
CC: ***@***.***>; ***@***.***>
Asunto: Re: [chiztechnology/Israel-Portfolio] Set client side validation (PR #13)

STATUS: INVALID 🔴

Hi @maherramadan78<https://github.com/maherramadan78>
This PR is invalid because it's your partner's PR😓kindly work on your repository and submit YOUR pull request 👍

Best of luck!


Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.


As described in the Code reviews limits policy<https://microverse.zendesk.com/hc/en-us/articles/1500004088561> you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this form<https://airtable.com/shrQAqnBwek5a0O0s>.

**Comment by ha-manel** [comment]
@maherramadan78 Please check the submission mode in the project requirements.

image
image
image

**Pull request merged**